### PR TITLE
adding quantization.rst file for quantization feature

### PR DESCRIPTION
**Pull request by gottbrath**
This was written by Raghu, Jessica, Dmytro and myself.

This PR will accumulate additional changes (there are a few more things we need to add to this actual rst file). I'll probably add the related image files to this PR as well. 

I'm breaking draft PR https://github.com/pytorch/pytorch/pull/27553 into more easily digestible pieces.

### CODE REVIEW

**Comment by gottbrath** [comment]
after this we would want https://github.com/pytorch/pytorch/pull/27603 to land (adding quantization into the index, etc..)

File: docs/source/quantization.rst [hunk 1]
```diff
@@ -0,0 +1,548 @@
...
+* *linear* — Linear (fully-connected) op
+* *max_pool2d* — 2D max pooling
+* *relu* — Rectified linear unit
+* *upsample* — Upsampler. Will be deprecated in favor of :func:`~torch.nn.quantized.functional.interpolate`
+* *upsample_bilinear* — Bilenear upsampler. Will be deprecated in favor of :func:`~torch.nn.quantized.functional.interpolate`
+* *upsample_nearest* — Nearest neighbor upsampler. Will be deprecated in favor of :func:`~torch.nn.quantized.functional.interpolate`
+
+**torch.ops.quantized**
+
+* TBD
```
Comment (jlin27) [inline docs/source/quantization.rst]: Missing section

File: docs/source/quantization.rst [hunk 2]
```diff
@@ -0,0 +1,548 @@
...
+* *relu* — Rectified linear unit
+* *upsample* — Upsampler. Will be deprecated in favor of :func:`~torch.nn.quantized.functional.interpolate`
+* *upsample_bilinear* — Bilenear upsampler. Will be deprecated in favor of :func:`~torch.nn.quantized.functional.interpolate`
+* *upsample_nearest* — Nearest neighbor upsampler. Will be deprecated in favor of :func:`~torch.nn.quantized.functional.interpolate`
+
+**torch.ops.quantized**
+
+* TBD  
+
+**NOTE: still TBD here**
```
Comment (jlin27) [inline docs/source/quantization.rst]: Missing section

File: docs/source/quantization.rst [hunk 3]
```diff
@@ -0,0 +1,548 @@
...
+
+Quantization workflows work by adding (e.g. adding observers as
+``.observer`` submodule) or replacing (e.g. converting ``nn.Conv2d`` to
+``nn.quantized.Conv2d``) submodules in the model's module hierarchy. It
+means that the model stays a regular ``nn.Module``-based instance throughout the
+process and thus can work with the rest of PyTorch APIs.
+
+
+Model preparation for quantization
+---------------------------------
```
Comment (jlin27) [inline docs/source/quantization.rst]: Capitalization "Model Preparation for Quantization"

File: docs/source/quantization.rst (hunk 2 above)
Comment (gottbrath) [reply docs/source/quantization.rst]: yea, this is just one section..

**Comment by gottbrath** [comment]
great. these comments look good. Will try to resolve ASAP.

File: docs/source/quantization.rst (hunk 1 above)
Comment (gottbrath) [reply docs/source/quantization.rst]: yep.. it is getting filled in now.

**Review by jlin27 — APPROVED** [review]

**Pull request closed without merging**